src/main/java/de/picturesafe/search/samples/collapse/Collapse.java

```diff
@@ -0,0 +1,106 @@
+import de.picturesafe.search.elasticsearch.model.SearchResult;
+import de.picturesafe.search.expression.Expression;
+import de.picturesafe.search.expression.ValueExpression;
+import de.picturesafe.search.parameter.CollapseOption;
+import de.picturesafe.search.parameter.InnerHitsOption;
+import de.picturesafe.search.parameter.SearchParameter;
+import de.picturesafe.search.parameter.SortOption;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.AnnotationConfigApplicationContext;
+import org.springframework.context.annotation.ComponentScan;
+import org.springframework.stereotype.Component;
+
+import java.util.Arrays;
+
+@Component
+@ComponentScan
+public class Collapse {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(Collapse.class);
+
+    @Autowired
+    private SingleIndexElasticsearchService singleIndexElasticsearchService;
+
+    public static void main(String[] args) {
+        try (AnnotationConfigApplicationContext ctx = new AnnotationConfigApplicationContext(Collapse.class)) {
+            final Collapse collapse = ctx.getBean(Collapse.class);
+            collapse.run();
+        }
+    }
+
+    private void run() {
+        try {
+            singleIndexElasticsearchService.createIndexWithAlias();
+
+            singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING, Arrays.asList(
+                    DocumentBuilder.id(1).put("user", "Jane Doe").put("tweet", "Tweet A").put("likes", 10).build(),
+                    DocumentBuilder.id(2).put("user", "Jane Doe").put("tweet", "Tweet B").put("likes", 101).build(),
+                    DocumentBuilder.id(3).put("user", "John Doe").put("tweet", "Tweet C").put("likes", 8).build(),
+                    DocumentBuilder.id(4).put("user", "John Doe").put("tweet", "Tweet D").put("likes", 99).build(),
+                    DocumentBuilder.id(5).put("user", "Jane Doe").put("tweet", "Tweet E").put("likes", 5).build(),
+                    DocumentBuilder.id(6).put("user", "Jane Doe").put("tweet", "Tweet F").put("likes", 16).build()
+            ));
+
+            final Expression expression = new ValueExpression("tweet", ValueExpression.Comparison.LIKE, "Tweet");
+
+            findTweetsSortByLikes(expression);
+            findMostLikedTweetOfEachUserSortByLikes(expression);
+            findMostLikedTweetsOfEachUserSortByLikes(expression, 2);
+            findMostLikedTweetsOfEachUserSortByLikes(expression, 3);
+        } finally {
+            singleIndexElasticsearchService.deleteIndexWithAlias();
+        }
+    }
+
+    private void findTweetsSortByLikes(Expression expression) {
+        final SearchParameter searchParameter = SearchParameter.builder().sortOptions(SortOption.desc("likes")).build();
+        search(expression, searchParameter);
+    }
+
+    private void findMostLikedTweetOfEachUserSortByLikes(Expression expression) {
```

Comment:
1. There is only one user, so this example may not be the best.
2. Doesn't "most liked" imply that it's sort by likes? Why doubling it in the method name?

Reply:
There are two users: "Jane Doe" and "John Doe" ;-)

The users are sorted by most likes AND the document with the highest like is returned (for each user).
For example, if the users were sorted by name (user), the method would be 'findMostLikedTweetOfEachUserSortByUser'.

So it's okay for me.

Reply:
Hmm, the names look very similar, that's why i didn't see the difference. Wouldn't it be better to have obviously different names?

Reply:
Do you have a good idea? How about:

findTweetsSortByLikes
collapseMostLikedTweetForEachUser
expandMostLikedTweetsOfEachUser

Reply:
OK, i should have made clearer that i mean the user names. Sorry!

What about "John Doe" and "Jeanne d’Arc"

Reply:
ok, done!